src/classes/RD2_OpportunityEvaluationService.cls

```diff
@@ -97,11 +97,15 @@ public inherited sharing class RD2_OpportunityEvaluationService {
     }
 
     /**
-     * @description Beginning of the current calendar or fiscal year, whichever is earlier.
+     * @description Beggining of the current calendar or fiscal year, whichever is earlier.
+     * The start date is offset by the Next Donation Date match days before the year start date
+     * so past Opportunities matching future projected installments are returned.
     */
     private Date startDate {
         get {
-            return yearValueProcessor.getStartDate();
+            return yearValueProcessor
+                .getStartDate()
+                .addDays(-rdSettings.nextDonationDateMatchDays);
```

Comment:
Good question, this `startDate` statement above is used to retrieve Opps within the date range of the calendar/fiscal year. When the past calendar or fiscal year Opp is matching the next year installment, then it is included in the Current Year Value and its installment is excluded from the Next Year Value. When the next year arrives, the installment matching the past year Opp is excluded from the Current Year Value. This [test](https://github.com/SalesforceFoundation/NPSP/blob/aa3eb70b1b63be67693eebe91a0d91dff16c69cc/src/classes/RD2_OpportunityEvaluationService_TEST.cls#L1262) is validating the scenario.

Reply:
After testing, I don't think it's working that way. 

Here's the scenario:
- RD is every 1 Month on the Last Day of the Month starting on 12/15/2019 for $100
- RD has Open installments on 12/31/2019, 1/31/2020, 2/29/2020, and 3/31/2020
- RD Statistics has $1200 for Current and Next Year
- Edit the 12/31 installment and change the CloseDate to 1/1/2020 (one day in the future) and set the stage to Closed/Won
- Expected: Current Year value is still $1200 (because the 1/1/2020 opp is the 12/31 installment)
- Actual: Current Year value is $1300.

@mpusto Can you confirm that the above is what you would expect as well?

Reply:
@heidmary ^

Reply:
@force2b , I cannot reproduce the behaviour. When the Opp Close Date = Dec 31, 2019, then the Current Year Value = $1,100. When the Opp Close Date is set to be Jan 1 and the Opp is Closed, the Current Year Value = $1,200.